pkg/cloud/vsphere/services/govmomi/esxi/clone.go

```diff
@@ -17,12 +17,271 @@ limitations under the License.
 package esxi
 
 import (
+	"fmt"
+	"path"
+	"strings"
+
 	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
 
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/context"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/services/govmomi/extra"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/services/govmomi/template"
 )
 
 // Clone kicks off a clone operation on ESXi to create a new virtual machine.
 func Clone(ctx *context.MachineContext, userData []byte) error {
-	return errors.New("temporarily disabled esxi support")
+	vm, err := NewVmContext(ctx)
```

Comment:
Hi @moshloop,

Maybe this is `NewESXiLoggerContext`? Or is this sort of a combination of the VM builder you previously constructed?